File: frontend/src/types/intervention.ts
/**
 * Intervention interfaces - Agentic decision outputs
 */

export type ActionType =
  | 'suggest_walk'
  | 'suggest_balance_exercise'
  | 'suggest_strength_exercise'
  | 'suggest_stretching'
  | 'suggest_rest'
  | 'suggest_hydration'
  | 'alert_caregiver'
  | 'recommend_doctor'
  | 'emergency_alert'
  | 'encourage_activity'
  | 'celebrate_progress'
  | 'request_assessment'
  | 'continue_monitoring';

export interface InterventionAction {
  user_id: string;
  timestamp: string;
  action_type: ActionType;
  priority: number; // 1-5
  raw_message: string;
  localized_message?: string;
  trigger_reason: string;
  suggested_duration_minutes?: number;
  exercise_video_url?: string;
}

File: backend/app/models/intervention.py
"""Intervention models - Agentic decision outputs."""

from enum import Enum
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class ActionType(str, Enum):
    """
    The "Agent" decision outputs.
    Based on Observe-Orient-Decide-Act loop.
    """
    # Exercise suggestions
    SUGGEST_WALK = "suggest_walk"
    SUGGEST_BALANCE_EXERCISE = "suggest_balance_exercise"
    SUGGEST_STRENGTH_EXERCISE = "suggest_strength_exercise"
    SUGGEST_STRETCHING = "suggest_stretching"

    # Rest/Recovery
    SUGGEST_REST = "suggest_rest"
    SUGGEST_HYDRATION = "suggest_hydration"

    # Alerts
    ALERT_CAREGIVER = "alert_caregiver"
    RECOMMEND_DOCTOR = "recommend_doctor"
    EMERGENCY_ALERT = "emergency_alert"

    # Motivation
    ENCOURAGE_ACTIVITY = "encourage_activity"
    CELEBRATE_PROGRESS = "celebrate_progress"

    # Monitoring
    REQUEST_ASSESSMENT = "request_assessment"
    CONTINUE_MONITORING = "continue_monitoring"


class InterventionAction(BaseModel):
    """
    Output from the agentic decision system.
    Will be translated via SeaLion for user-facing display.
    """
    user_id: str
    timestamp: datetime = Field(default_factory=datetime.now)

    action_type: ActionType
    priority: int = Field(default=1, ge=1, le=5, description="1=low, 5=urgent")

    # Raw message (to be translated by SeaLion)
    raw_message: str = Field(..., description="Generic advice before SeaLion translation")

    # Translated message (after SeaLion processing)
    localized_message: Optional[str] = Field(None, description="Singlish/Dialect translated message")

    # Context
    trigger_reason: str = Field(..., description="Why this action was triggered")
    suggested_duration_minutes: Optional[int] = None
    exercise_video_url: Optional[str] = None
